src/Network/NWConnectionGroup.cs

```diff
@@ -0,0 +1,276 @@
+				if (x == IntPtr.Zero)
+					return null;
+				return new NWParameters (x, owns: true);
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_connection_group_start (OS_nw_connection_group group);
+
+		public void Start () => nw_connection_group_start (GetCheckedHandle ()); 
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_connection_group_cancel (OS_nw_connection_group group);
+
+		public void Cancel () => nw_connection_group_cancel (GetCheckedHandle ()); 
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_connection_group_set_queue (OS_nw_connection_group group, IntPtr queue);
+
+		public void SetQueue (DispatchQueue queue)
+		{
+ 			if (queue == null)
+				throw new ArgumentNullException (nameof (queue));
+
+			nw_connection_group_set_queue (GetCheckedHandle (), queue.Handle);
+		}
+
+		// can return null
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_endpoint nw_connection_group_copy_local_endpoint_for_message (OS_nw_connection_group group, OS_nw_content_context context);
+
+		public NWEndpoint? GetLocalEndpoint (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_copy_local_endpoint_for_message (GetCheckedHandle (), context.Handle);
+			return ptr == IntPtr.Zero ? null : new NWEndpoint (ptr, owns: true);
+		}
+
+		// can return null
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_path nw_connection_group_copy_path_for_message (OS_nw_connection_group group, OS_nw_content_context context);
+
+		public NWPath? GetPath (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_copy_path_for_message (GetCheckedHandle (), context.Handle);
+			return ptr == IntPtr.Zero ? null : new NWPath (ptr, owns: true);
+		}
+
+		// can return null
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_endpoint nw_connection_group_copy_remote_endpoint_for_message (OS_nw_connection_group group, OS_nw_content_context context);
+
+		public NWEndpoint? GetRemmoteEndpoint (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_copy_remote_endpoint_for_message (GetCheckedHandle (), context.Handle);
+			return ptr == IntPtr.Zero ? null : new NWEndpoint (ptr, owns: true);
+		}
+
+		// can return null
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_connection nw_connection_group_extract_connection_for_message (OS_nw_connection_group group, OS_nw_content_context context);
+
+		public NWConnection? GetConnection (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_extract_connection_for_message (GetCheckedHandle (), context.Handle);
```

Comment:
Use `GetCheckedHandle` so you don't end up providing a non-null managed instance that is disposed.

IOW when the instance is disposed then `Handle` is `IntPtr.Zero`. You would effectively call the native API with `nil`, which is not allowed.